File: go/internal/replay/parserfixture/testdata/trees/dart/order_service.dart
import 'order_repository.dart';

class OrderService {
  final OrderRepository repository;

  OrderService(this.repository);

  Order place(String customerId, List<String> skus) {
    final order = repository.create(customerId);
    for (final sku in skus) {
      order.addLine(sku);
    }
    return repository.save(order);
  }

  Order find(String orderId) => repository.byId(orderId);
}

File: go/internal/replay/parserfixture/testdata/trees/java/OrderService.java
package com.example.orders;

import java.util.List;

/** OrderService coordinates order placement and lookup. */
public class OrderService {
    private final OrderRepository repository;

    public OrderService(OrderRepository repository) {
        this.repository = repository;
    }

    public Order place(String customerId, List<String> skus) {
        Order order = repository.create(customerId);
        for (String sku : skus) {
            order.addLine(sku);
        }
        return repository.save(order);
    }

    public Order find(String orderId) {
        return repository.byId(orderId);
    }
}
